BentoKit/BentoKit/Components/Search.swift

```diff
@@ -0,0 +1,126 @@
+import Bento
+import StyleSheets
+
+extension Component {
+    public final class Search: AutoRenderable {
+        public let configurator: (View) -> Void
+        public let styleSheet: StyleSheet
+
+        public init(
+            placeholder: String? = nil,
+            keyboardType: UIKeyboardType = .default,
+            didBeginEditing: Optional<(UISearchBar) -> Void> = nil,
+            textDidChange: Optional<(UISearchBar, String) -> Void> = nil,
+            cancelButtonClicked: Optional<(UISearchBar) -> Void> = nil,
+            styleSheet: StyleSheet = StyleSheet()
+            ) {
+            self.configurator = { view in
+                view.searchBar.placeholder = placeholder
+                view.searchBar.keyboardType = keyboardType
+                view.didBeginEditing = didBeginEditing
+                view.textDidChange = textDidChange
+                view.cancelButtonClicked = cancelButtonClicked
+            }
+            self.styleSheet = styleSheet
+        }
+    }
+}
+
+extension Component.Search {
+    public final class View: BaseView, UISearchBarDelegate {
+
+        let searchBar = UISearchBar()
+
+        var textDidChange: Optional<(UISearchBar, String) -> Void> = nil
```

Comment:
Do we really need the `UISearchBar` there?

Reply:
I decided to follow the signature of the delegate method. also it makes it easier to do things like dismiss keyboard when cancel is pressed (for some reason is not a default behaviour) without jumping through the hoops.

Reply:
fair enough